utils.go

```diff
@@ -57,3 +58,30 @@ func parseImageDestination(c *cli.Context, name string) (types.ImageDestination,
 	}
 	return nil, fmt.Errorf("Unrecognized image reference %s", name)
 }
+
+func parseDockerImage(img, certPath string, tlsVerify bool) (types.Image, error) {
+	s, err := newDockerImageSource(img, certPath, tlsVerify)
+	if err != nil {
+		return nil, err
+	}
+	return &dockerImage{src: s}, nil
+}
+
+// parseDockerImageName converts a string into a reference and tag value.
+func parseDockerImageName(img string) (reference.Named, string, error) {
```

Comment:
so what if we hide this parse login directly into `newDockerClient` (so it calls `parseDockerImageName` and `newDockerClient` just accepts a `string`?